build.gradle

```diff
@@ -220,6 +220,7 @@ project(':iceberg-core') {
     compile "com.fasterxml.jackson.core:jackson-databind"
     compile "com.fasterxml.jackson.core:jackson-core"
     compile "com.github.ben-manes.caffeine:caffeine"
+    compile "org.rocksdb:rocksdbjni"
```

Comment:
I know this is a WIP, but is it possible to avoid bringing this into the class path once this is done?

I believe Flink brings in their own RocksDB fork (frocksdb maybe?) and I imagine others might too.

Reply:
As the `StructLikeMap` is in the `core` code path, which don't depend on any specific compute engines(spark/flink/hive/presto etc), so in theory we could not assume that the engine runtime will include this dependency jar.  It's better to include it in iceberg's jar if possible.

Reply:
That makes sense.

I'd advocate for possibly sharing it though, so that multiple versions can exist together.

With Spark 3.2 bringing support for RocksDB statestore as well, it might be wise to shade it in the finl outcome. But that is admittedly a long ways off.